src/v/cluster/data_policy_manager.h

```diff
@@ -37,6 +40,7 @@ class data_policy_manager {
 
 private:
     ss::sharded<v8_engine::data_policy_table>& _dps;
+    std::unique_ptr<v8_engine::api>& _v8_engine_api;
```

Comment:
let's avoid storing a reference to a unique_ptr its kind of an anti-pattern

Reply:
I made data_policy_manager which handle data_policy events from log.
If v8_engine is not enabled we also need to handle this events from log, but do not need to do some work about compilation v8_engine.
So I need to pass ref to unique_pointer, or sharded_services (but we want to store this sharded_services inside v8::api), or optional<smth&> (looks like it is anti-pattern too), or aw pointer (we try to avoid this too), because in data_policy_manager I check v8 flag from config and can do or not some work. 

So I am a little bit confused and do not know what is the right solution